lib/compute/index.js

```diff
@@ -50,17 +64,17 @@ exports.image = function image(id, payload) {
         return reject(err);
       }
       var buf = Buffer.concat(arr);
-      imageCc.enqueue({
-        id: id,
-        payload: buf
-      }, function(err, res) {
-        if (err) {
-          logger.error('process.error', err);
-          reject(AppError.processingError(err));
-        } else {
-          resolve(res);
-        }
-      });
+
+      resolve(P.all(Object.keys(SIZES).map(function(variant) {
+        var hw = SIZES[variant];
+        return enqueue({
+          id: id,
+          suffix: (variant === 'default') ? '' : variant,
+          height: hw.h,
+          width: hw.w,
+          payload: buf
+        });
+      })));
```

Comment:
Gross indeed, but I think worthwhile.  It may require extract some higher-level "task" abstraction here to keep things sane.